AirCasting/SessionViews/ChartDatabaseObserver.swift

```diff
@@ -33,27 +33,61 @@ final class ChartDatabaseObserver {
         guard let insertedObjects = (n.userInfo?[NSInsertedObjectsKey] as? NSSet)?.allObjects else { return }
         let insertedMeasurements = insertedObjects.compactMap { $0 as? MeasurementEntity }
         let filtered = insertedMeasurements.filter { $0.measurementStream.session.uuid == session && $0.measurementStream.sensorName == sensor }
-        guard filtered.count > 0, shouldFireObserver() else { return }
-        lastMeasurementChangeAt = DateBuilder.getRawDate()
+        guard let newestMeasurement = filtered.sorted(by: { $0.time > $1.time }).first else { return }
+        defer { latestMeasurementTime = newestMeasurement.time }
+        guard filtered.count > 0, shouldFireObserver(newMeasurement: newestMeasurement) else { return }
+        Log.info("Firing.\n" +
+                 "   New measurement time: \(newestMeasurement.time.debugInfo),\n" +
+                 "   session name: \(newestMeasurement.measurementStream.session.name ?? "??")\n" +
+                 "   session start time: \(newestMeasurement.measurementStream.session.startTime.debugInfo),\n" +
+                 "   filtering: \(filteringComponent)")
         onMeasurementsChange()
     }
     
-    private func shouldFireObserver() -> Bool {
-        guard let lastMeasurementChangeAt = lastMeasurementChangeAt, let filteringComponent = filteringComponent else { return true }
-        let now = DateBuilder.getRawDate()
-        guard Calendar.current.isDate(lastMeasurementChangeAt, inSameDayAs: now) else { return true }
-        let nowComponent = Calendar.current.component(filteringComponent, from: now)
-        let lastMeasurementComponent = Calendar.current.component(filteringComponent, from: lastMeasurementChangeAt)
-        return lastMeasurementComponent != nowComponent
+    private func shouldFireObserver(newMeasurement: MeasurementEntity) -> Bool {
+        switch filteringComponent {
+        case .none: return true
+        case .hour: return !isInNewHour(measurement: newMeasurement)
+        case .minute(let countingFrom): return isInNewMinute(measurement: newMeasurement, countingFrom: countingFrom)
+        }
+    }
+    
+    private func isInNewHour(measurement: MeasurementEntity) -> Bool {
+        guard let latestMeasurementTime = latestMeasurementTime else { return true }
+        guard Calendar.current.isDate(latestMeasurementTime, inSameDayAs: measurement.time) else { return true }
+        let nowComponent = Calendar.current.component(.hour, from: measurement.time)
+        let latestComponent = Calendar.current.component(.hour, from: latestMeasurementTime)
+        return latestComponent == nowComponent
```

Comment:
Based on name shouldn't it be `!=`?

Reply:
good catch, changed it (and corresponding call) to match the minute specific func naming